algebird-core/src/main/scala/com/twitter/algebird/MomentsGroup.scala

```diff
@@ -24,36 +24,84 @@ import algebra.{CommutativeGroup, CommutativeMonoid}
  * and kurtosis.
  *
  * m{i} denotes the ith central moment.
+ *
+ * This code manually inlines code to make it look like a case class. This is done
+ * because we changed the count from a Long to a Double to enable the scale method,
+ * which allows exponential decays of moments, but we didn't want to break backwards
+ * binary compatibility.
  */
-case class Moments(m0: Long, m1: Double, m2: Double, m3: Double, m4: Double) {
+sealed class Moments(val m0D: Double, val m1: Double, val m2: Double, val m3: Double, val m4: Double)
+    extends Product
+    with Serializable {
+  def this(m0: Long, m1: Double, m2: Double, m3: Double, m4: Double) {
+    this(m0.toDouble, m1, m2, m3, m4)
+  }
+
+  def m0: Long = m0D.toLong
+
   def count: Long = m0
 
+  def totalWeight: Double = m0D
+
   def mean: Double = m1
 
   // Population variance, not sample variance.
   def variance: Double =
-    if (count > 1)
-      m2 / count
-    else
-      /* don't return junk when the moment is not defined */
-      Double.NaN
+    m2 / m0D
 
   // Population standard deviation, not sample standard deviation.
   def stddev: Double = math.sqrt(variance)
 
   def skewness: Double =
-    if (count > 2)
-      math.sqrt(count) * m3 / math.pow(m2, 1.5)
-    else
-      /* don't return junk when the moment is not defined */
-      Double.NaN
+    math.sqrt(m0D) * m3 / math.pow(m2, 1.5)
```

Comment:
No longer filtering on `count > 2` is a semantic change that's probably worth at least calling out in the release notes or something.

Reply:
indeed it is. I will call it out, but since we returned NaN before, we were already in a "silent-ish" failure more.